File: JANUARY/07-01-2026/Count distinct elements in every window.py
class Solution:
    def countDistinct(self, arr, k):
        from collections import defaultdict
        
        freq = defaultdict(int)
        res = []
        
        for i in range(len(arr)):
            freq[arr[i]] += 1
            
            if i >= k - 1:
                res.append(len(freq))
                
                left = arr[i - k + 1]
                freq[left] -= 1
                if freq[left] == 0:
                    del freq[left]
                    
        return res

File: JANUARY/07-01-2026/Count distinct elements in every window.cs
using System;
using System.Collections.Generic;

class Solution {
    public List<int> countDistinct(int[] arr, int k) {
        List<int> res = new List<int>();
        Dictionary<int, int> freq = new Dictionary<int, int>();
        
        for (int i = 0; i < arr.Length; i++) {
            if (freq.ContainsKey(arr[i]))
                freq[arr[i]]++;
            else
                freq[arr[i]] = 1;
            
            if (i >= k - 1) {
                res.Add(freq.Count);
                
                int left = arr[i - k + 1];
                freq[left]--;
                if (freq[left] == 0)
                    freq.Remove(left);
            }
        }
        
        return res;
    }
}